### PR TITLE
Reopen subscriptions after error/close events from subscriber

### PR DESCRIPTION
**Pull request by tcichowicz**
Sometimes our subscriptions stopped receiving messages after receiving an error event:
```
{
  fullError: "StatusError: Call cancelled (...)",
  code: 1,
  details: "Call cancelled",
  level: 500   
  message: "Error: Call cancelled (...)"   
 }
```

I'm not 100% sure this PR will fix it 😅 Unfortunately I could not pin-point the root cause. But it's at least some attempt at a workaround. I've upgraded the dependency as well, maybe it was fixed on the library side. Let's try this if you deem the change safe and then observe if we handle these errors correctly.

### CODE REVIEW

File: packages/pubsub/src/Subscriber.ts [hunk 1]
```diff
@@ -101,6 +102,23 @@ export class Subscriber<T = unknown> {
...
+    await this.subscription.close()
+    this.subscription.open()
+    logger.info('Reopened subscription after error', {
+      error,
+      name: this.subscription.name
+    })
+  }
+
+  private processClose = () => {
+    this.subscription.open()
```
Comment (kirpichenko) [inline packages/pubsub/src/Subscriber.ts:116]: Are you sure? Subscription will be never closed?

Comment (tcichowicz) [reply packages/pubsub/src/Subscriber.ts:116]: You mean to try to close the subscription anyway?

Comment (tcichowicz) [reply packages/pubsub/src/Subscriber.ts:116]: I based this on the presumption that `close` event can only be emitted:
- by subscription passed from subscriber: https://github.com/googleapis/nodejs-pubsub/blob/master/src/subscription.ts#L271
- by subscriber when `close()` is called: https://github.com/googleapis/nodejs-pubsub/blob/master/src/subscriber.ts#L316

**Comment by tcichowicz** [comment]
@kirpichenko I already published the package:
https://www.npmjs.com/package/@join-com/pubsub/v/0.0.27
Please merge the branch as I don't have access.

**Pull request merged**